### PR TITLE
Nina debug squared dist

### PR DESCRIPTION
**Pull request by ninamiolane**
More tests and result of debugging of the squared distance

### CODE REVIEW

File: geomstats/special_orthogonal_group.py [hunk 1]
```diff
@@ -40,14 +40,31 @@ def skew_matrix_from_vector(vec):
...
+
+    :param skew_mat: 3x3 skew-symmetric matrix
+    :return vec: 3d vector
+    """
+    assert skew_mat.shape == (3, 3)
+
+    vec = np.array([skew_mat[2][1],
+                    skew_mat[0][2],
+                    skew_mat[1][0]])
+    return vec
```
Comment (johmathe) [inline geomstats/special_orthogonal_group.py]: return skew_mat[(2,0,1), (1,2,0)] ?

File: geomstats/special_orthogonal_group.py [hunk 2]
```diff
@@ -86,72 +103,71 @@ def regularize(self, rot_vec):
...
+
+        trace = np.trace(rot_mat)
+        cos_angle = .5 * (trace - 1)
+        cos_angle = np.clip(cos_angle, -1, 1)
         angle = np.arccos(cos_angle)
 
-        # -- Edge case: angle close to 0
+        rot_vec = vector_from_skew_matrix(rot_mat - rot_mat.transpose())
+
         if angle < epsilon:
```
Comment (johmathe) [inline geomstats/special_orthogonal_group.py]: nit: as I see these, I'm thinking that maybe having a function if IsClose(angle, 0) or if IsClose(angle, pi) with Isclose being something like return np.abs(a-b) < epsilon would be nice?

File: tests/helper.py [hunk 3]
```diff
@@ -33,19 +35,19 @@ def exp_then_log_from_identity(metric, tangent_vec):
 # Standard
 
 
-def log_then_exp(metric, base_point, point):
-    aux = metric.log(base_point=base_point,
-                     point=point)
-    result = metric.exp(base_point=base_point,
-                        tangent_vec=aux)
+def log_then_exp(metric, point, base_point):
+    aux = metric.log(point=point,
```
Comment (johmathe) [inline tests/helper.py]: this shouldn't matter for named arguments.

File: tests/helper.py [hunk 4]
```diff
@@ -56,31 +58,48 @@ def exp_then_log(metric, base_point, tangent_vec):
...
-                             point=aux)
+def group_exp_then_log(group, tangent_vec, base_point):
+    aux = group.group_exp(tangent_vec=tangent_vec,
+                          base_point=base_point)
+    result = group.group_log(point=aux,
+                             base_point=base_point)
     return result
+
+
+# Regularize a tangent_vector by getting its norm,
```
Comment (johmathe) [inline tests/helper.py]: comment should be in the docstring of the function, not here.

File: tests/helper.py [hunk 5]
```diff
@@ -56,31 +58,48 @@ def exp_then_log(metric, base_point, tangent_vec):
...
 
 def group_exp_then_log_from_identity(group, tangent_vec):
-    aux = group.group_exp(tangent_vec)
-    result = group.group_log(aux)
+    aux = group.group_exp_from_identity(tangent_vec=tangent_vec)
+    result = group.group_log_from_identity(point=aux)
     return result
 
 
 # Standard
```
Comment (johmathe) [inline tests/helper.py]: put this comment in the docstring too

File: geomstats/special_euclidean_group.py [hunk 6]
```diff
@@ -157,10 +158,14 @@ def group_exp_from_identity(self,
         :param base_point: 6d vector element of SE(3).
         :returns group_exp_transfo: 6d vector element of SE(3).
         """
+        # Regularize the tangent vector at the identity, because:
+        # - shooting with a vector or its regularized version
+        # gives the same point
+        # - need the regularized angle for the edge cases
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/special_euclidean_group.py]: Woups, sorry. Leftover :)

**Pull request merged**